057-run-zk-kraft-clusters-parallel.md

```diff
@@ -0,0 +1,65 @@
+There is no way to differentiate between clusters running in ZooKeeper or KRaft mode.
+When a `Kafka` custom resource is configured to actually use ZooKeeper or it's just badly configured but supposed to be KRaft based, the operator detects it as having a missing KRaft controllers configuration and logs the following warning:
+
+```shell
+io.strimzi.operator.common.model.InvalidResourceException: Tha Kafka cluster my-cluster is invalid: [At least one KafkaNodePool with the controller role and at least one replica is required when KRaft mode is enabled]
+```
+
+In this case, the reconciliation fails and the cluster is not operated at all.
+It means that the operator doesn't take any actions on changes applied by the user to the `spec.kafka` or `spec.zookeeper` in the `Kafka` custom resource.
+
+Furthermore, the `UseKRaft` feature gate was created for development and testing.
+As we move forward, it will be soon available for regular clusters and the current state is not going to be sufficient anymore.
+
+## Motivation
+
+Leaving a ZooKeeper-based cluster not operated when the `UseKRaft` feature gate is enabled looks like a bad thing for the user, until the ZooKeeper support will be completely removed.
+Before that, the user should be able to have both ZooKeeper and KRaft based clusters running and operated when the `UseKRaft` feature gate is enabled.
+Also, taking into account the future ZooKeeper to KRaft migration process, as described in the proposal [PR#90](https://github.com/strimzi/proposals/pull/90), it would help the operator migration component to detect which cluster is ZooKeeper-based so "migrate-able" and which one is already KRaft-based.
+
+## Proposal
+
+The proposal is about adding a new `strimzi.io/kraft` annotation to the `Kafka` custom resource, to be applied by the user, in order to identify a ZooKeeper or KRaft based cluster.
+The possible values would be:
+
+* `disabled`, missing or any invalid value: identifies a ZooKeeper-based cluster.
+* `enabled`: identifies a KRaft-based cluster.
+
+This way, during the reconciliation, the operator is able to "detect" a ZooKeeper-based cluster avoiding the warning and allowing the user to operate it.
+On the KRaft side, the annotation would be needed to have the operator reconciling the corresponding `Kafka` custom resource.
+Without the annotation, but the `UseKRaft` feature gate enabled, the operator would try to handle it as a ZooKeeper-based one.
+The `UseKRaft` feature gate does currently not support any upgrades of KRaft clusters.
+It is expected to be used only for short-lives clusters used for development and testing.
+So no clusters are expected to exist.
+Therefore adding the annotation now does not present any backwards compatibility issues.
+Only ZooKeeper-based clusters are expected and newly created KRaft-based clusters having the `strimzi.io/kraft: enabled`.
```

Comment:
One point with a consistency somewhere you use KRaft-based and somewhere Kraft based. I would stick with KRaft-based everywhere. But it's just a nit ).

Reply:
You are right but when it's just one I use "ZooKeeper-based cluster and KRaft-based cluster", when I want to mention both using plural I say "...ZooKeeper and KRaft based clusters...". Not sure if and where I should put `-based` because `based` is referring to both.